src/metabase/domain_entities/table.cljc

```diff
@@ -0,0 +1,27 @@
+(ns metabase.domain-entities.table
+  "Shared logic for manipulating Tables."
+  (:require
+   [metabase.util.malli :as mu]))
+
+;; Table IDs are either the database ID or a string "card__17".
+(defmacro projections
+  "Helper for generating the trivial \"projection\" functions that take an entity (a Clojure map) and extract one of
+  its fields."
+  [input-schema & specs]
+  `(do
+     ~@(for [[name implementation output-schema] specs
+             :let [arg-name (gensym "entity")]]
+         `(mu/defn ^:export ~name :- ~output-schema
+            ~(str "Projection function for " name " field")
+            [~arg-name :- ~input-schema]
+            (~implementation ~arg-name)))))
+
+(def table [:map
+            [:id [:alt string? number?]]
+            [:db_id int?]
+            [:description {:optional true} string?]])
+
+(projections table
+             [id :id [:alt string? int?]]
```

Comment:
Same concern about `:alt` vs `:or` here. If you mean it to be a vector with 1 string or int then this is correct.

Reply:
Same answer; this file is a ghost.